libvast/src/system/index.cpp

```diff
@@ -891,6 +891,8 @@ index(index_actor::stateful_pointer<index_state> self,
       // TODO: Consider switching layouts to a robin map to take advantage of
       // transparent key lookup with string views, avoding the copy of the name
       // here.
+      if (self->state.is_shutting_down)
```

Comment:
Maybe we can reuse the flag from CAF to avoid the extra state?
```suggestion
      if (stage->is_shutting_down)
```
Should be possible if we capture the stage pointer (make sure it isn't an owning type though).

Reply:
A solution that actually works should look something like:
```c++
if ((stage->flags() & (caf::stream_manager::is_stopped_flag | caf::stream_manager::is_shutting_down_flag)) != 0)
```

Reply:
Is it the stage which is set up a couple of lines above? It does not have ->flags().

```cpp
   self->state.stage = detail::attach_notifying_stream_stage(
    self,
    /* continuous = */ true,
    [](caf::unit_t&) {
      // nop
    },
    [self](caf::unit_t&, caf::downstream<table_slice>& out, table_slice x) {
      VAST_ASSERT(x.encoding() != table_slice_encoding::none);
      if ((self->state.stage->flags()
           & (caf::stream_manager::is_stopped_flag
              | caf::stream_manager::is_shutting_down_flag))
          != 0)
```

Reply:
For future reference, there's a `getf()` function in the stage so you don't have to work with the raw `flags_` variable.